pootle/static/js/browser/components/BrowserTable.js

```diff
@@ -0,0 +1,328 @@
+ * AUTHORS file for copyright and authorship information.
+ */
+
+import cx from 'classnames';
+import React, { PropTypes } from 'react';
+
+import TextToggle from 'components/TextToggle';
+import TimeSince from 'components/TimeSince';
+import LastActivity from 'components/LastActivity';
+import { t } from 'utils/i18n';
+
+const ColoredNumber = React.createClass({
+    propTypes: {
+        n: React.PropTypes.number,
+    },
+
+    render: function() {
+        if (!this.props.n) {
+            return (<span className="zero">0</span>);
+        }
+        return (
+            <span>{this.props.n}</span>
+        );
+    }
+});
+
+const NumberPill = React.createClass({
+    propTypes: {
+        n: React.PropTypes.number,
+        url: React.PropTypes.string.isRequired,
+    },
+
+    render: function() {
+        if (!this.props.n) {
+            return (<span className="zero">0</span>);
+        }
+        return (
+            <a className="stats-data" href={this.props.url}>{this.props.n}</a>
+        );
+    }
+});
+
+const ProgressBar = React.createClass({
+    render: function() {
+        var total = this.props.total || 0;
+        var fuzzy = this.props.fuzzy || 0;
+        var trans = this.props.translated || 0;
+        var trans_p = total > 0 ? Math.round(trans / total * 100) : 100;
+        var fuzzy_p = total > 0 ? Math.round(fuzzy / total * 100) : 0;
+        var untrans_p = 100 - trans_p - fuzzy_p;
+
+        var title= '<span class="legend translated"></span><span class="value translated">'+trans_p+'</span>% translated<br/>'+
+            '<span class="legend fuzzy"></span><span class="value fuzzy">'+fuzzy_p+'</span>% needs work<br/>'+
+            '<span class="legend untranslated"></span><span class="value untranslated">'+untrans_p+'</span>% untranslated';
+
+        return (
+            <table className="graph" title={title}>
+                <tbody>
+                    <tr>
+                        <td className="translated" style={{'width': trans_p+'%'}}><span></span></td>
+                        <td className="fuzzy" style={{'width': fuzzy_p+'%'}}><span></span></td>
+                        <td className="untranslated"><span></span></td>
+                    </tr>
+                </tbody>
+            </table>
+        );
+    }
+});
+
+const BrowserTable = React.createClass({
+
+    propTypes: {
+        items: React.PropTypes.array.isRequired,
+    },
+
+    // TODO: is there a better way to define constants (not fields)?
+    COL_TITLE: 0,
+    COL_PROGRESS: 1,
+    COL_TOTAL: 2,
+    COL_LASTUPDATED: 3,
+    COL_CRITICAL: 4,
+    COL_SUGGESTIONS: 5,
+    COL_INCOMPLETE: 6,
+    COL_LASTACTIVITY: 7,
+
+    getInitialState : function() {
+        return {
+            showDisabledRows: false,
+            sortColumn: this.COL_TITLE,
+            reverseSort: false,
+        };
+    },
+
+    componentWillMount() {
+        // check if there are disabled items
+        this.hasDisabledItems = false;
```

Comment:
We need to make sure this is also calculated in `componentWillReceiveProps`, otherwise as data changes, we won't get an updated value for `hasDisabledItems`.